### PR TITLE
Add WithSession helper for Cassandra driver

### LINKED ISSUE #11: No WithInstance function in the Cassandra driver
Just had a need for it since the Open method does not support multiple hosts, and found out the driver doesn't ship one.

### PR DESCRIPTION
**Pull request by mixedCase**
Closes #11.

Also tweaks the method receivers to match the struct name.

### CODE REVIEW

File: database/cassandra/cassandra.go [hunk 1]
```diff
@@ -35,11 +36,33 @@ type Cassandra struct {
 	session  *gocql.Session
 	isLocked bool
 
-	// Open and WithInstance need to guarantee that config is never nil
+	// Open and WithSession need to guarantee that config is never nil
 	config *Config
 }
 
-func (p *Cassandra) Open(url string) (database.Driver, error) {
+func WithSession(session *gocql.Session, config *Config) (database.Driver, error) {
```
Comment (kenjones-cisco) [inline database/cassandra/cassandra.go]: for consistency with all the other drivers, shouldn't the name remain as `WithInstance`?

Comment (dhui) [reply database/cassandra/cassandra.go]: Agreed, use the name `WithInstance` for consistency

**Comment by mixedCase** [comment]
@dhui Added both requests. Travis still seems to be failing. I am unsure how these changes are affecting the tests, since (at least before the lock change) it's just a variable name change and a completely new function.

**Comment by dhui** [comment]
Yeah, I don't think your changes are related to the failing tests.

Thanks for the PR! I'll merge in your changes, but if you wanna use them, use the master branch as the failing tests are release blockers.

**Pull request merged**